### PR TITLE
Revamp: Get API Definition from enforcer

### PR DESCRIPTION
**Pull request by BLasan**
### Related PR
- https://github.com/wso2/apk/pull/1188

### CODE REVIEW

**Comment by BLasan** [comment]
Backoffice tests are passing
Screenshot 2023-06-17 at 09 43 41

File: test/integration/integration/tests/fetch-api-definition.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			{
+				Request: http.Request{
+					Host: "fetch-api-definition.test.gw.wso2.com",
+					Path: "/fetch-api-definition/v1.0.0?definitionType=swagger",
+					Headers: map[string]string{
+						"content-type": "application/json",
+					},
+					Method: "GET",
+				},
+				Response: http.Response{
```
[the remark this answers is not in the document]
Comment (BLasan) [reply test/integration/integration/tests/fetch-api-definition.go:51]: Fixed via [372b438](https://github.com/wso2/apk/pull/1266/commits/372b4386fa7c971ab071ac71e7c57fe492d2c199)

File: test/integration/integration/tests/fetch-api-definition.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	Test: func(t *testing.T, suite *suite.IntegrationTestSuite) {
+		ns := "gateway-integration-test-infra"
+		gwAddr := "fetch-api-definition.test.gw.wso2.com:9095"
+		token := http.GetTestToken(t)
+
+		testCases := []http.ExpectedResponse{
+			{
+				Request: http.Request{
+					Host: "fetch-api-definition.test.gw.wso2.com",
+					Path: "/fetch-api-definition/v1.0.0?definitionType=swagger",
```
[the remark this answers is not in the document]
Comment (BLasan) [reply test/integration/integration/tests/fetch-api-definition.go]: Fixed via [92ae43f](https://github.com/wso2/apk/pull/1266/commits/92ae43fd8a4293af9a58e25d338a97408f505bdc)

**Review by AmaliMatharaarachchi — APPROVED** [review]

File: test/integration/integration/tests/fetch-api-definition.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+				Response: http.Response{
+					StatusCode: 404,
+				},
+				Backend:      "infra-backend-v1",
+				Namespace:    ns,
+				TestCaseName: "FetchAPIDefinition",
+			},
+		}
+
+		for i := range testCases {
```
[the remark this answers is not in the document]
Comment (BLasan) [reply test/integration/integration/tests/fetch-api-definition.go:76]: Done

**Pull request merged**